pom.xml

```diff
@@ -66,6 +66,8 @@
                 <module>vaadin-bom</module>
                 <module>vaadin-core-internal</module>
                 <module>vaadin-core</module>
+                <module>vaadin-dev-flow</module>
+                <module>vaadin-dev-hilla</module>
```

Comment:
Not really a comment against this design, but just a concern how backwards compatible will this approach be: in existing projects the artifact `vaadin-dev` is excluded from prod builds. Now, as only `vaadin-dev-flow` is included into `vaadin` and `vaadin-core`, the old approach with excluding `vaadin-dev` will have no effect (I suppose Maven work in this way).

So we either have to document this clearly, or let vaadin-dev be semantically `vaadin-flow-dev`, which I don't really like.

Reply:
We should have a clear mention in the upgrade notes that it is named vaadin-dev-flow in 24.4

Also as the vaadin(-core) packages are getting "deprecated" it should be noted how the new package structure is going to be used in the update notes also.

Reply:
As we discussed offline, we can set the `hilla-dev` dependency as `optional` in `vaadin-dev`, which basically would relief us from the need of splitting `vaadin-dev` into two parts.
Then `vaadin` and `vaadin-core` wound't get Hilla dev things and in `vaadin-spring-boot-starter` we can explicitly add `hilla-dev` in the list of dependencies. 
Finally `vaadin-dev` exclusion should work in a backwards compatible way after that.

Reply:
> Also as the vaadin(-core) packages are getting "deprecated" it should be noted how the new package structure is going to be used in the update notes also.

Not sure that in 24.4 we are going to deprecate them, as we don't provide an alternative in 24.4.
I think once we introduce the new starter artifacts, then we should deprecate.